tests/engine/adhoc_test.go

```diff
@@ -71,16 +71,22 @@ func testFibonacci(t *testing.T, newEngine func() *wazero.Engine) {
 	// We execute 1000 times in order to ensure the JIT engine is stable under high concurrency
 	// and we have no conflict with Go's runtime.
 	const goroutines = 1000
+
```

Comment:
this change in testFibonacci verifies that the concurrent execution of module functions is thread-safe!